lib/storage/s3.js

```diff
@@ -1,142 +1,142 @@
 "use strict";
 
-class S3 {
-    /**
-     * @param {Object} awsSdk
-     * @param {Object} promise
-     */
-    constructor(awsSdk, promise) {
-        this.aws = awsSdk;
-        this.config = {};
-        this.configure({});
-        this.promise = promise;
-        this.awsS3 = null;
-    }
-
-
-    /**
-     * Configures options for AWS.
-     *
-     * @param {Object} config
-     * @return {Object} this
-     */
-    configure(config) {
-        this.aws.config.region = config.region || "us-east-1";
-        this.config.bucket = config.bucket || null;
-
-        return this;
-    }
+module.exports = function (awsSdk, promise) {
+    class S3 {
+        constructor(config) {
+            this.config = {};
+            this.configure(config);
+            this.awsS3 = null;
+        }
 
 
-    /**
-     * Deletes a single file from configured bucket.
-     *
-     * @param {string} filename
-     * @return {Promise.<*>} promise
-     */
-    delAsync(filename) {
-        return this.transit().deleteObjectAsync({
-            Key: filename
-        });
-    }
+        /**
+         * Configures options for AWS.
+         *
+         * @param {Object} config
+         * @return {Object} this
```

Comment:
Should this be something like `@return {S3} this` since it's returning the class?